Update LSP0 Pages (#461)
lukso-network/docs: docs/standards/smart-contracts/lsp0-erc725-account.md

Check the [**fallback(..)**](https://github.com/lukso-network/lsp-smart-contracts/blob/v0.8.0/contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol#L121) function implementation in **LSP0ERC725Account** Contract.

:::

```solidity
fallback() external payable
```

Executed when sending data to the contract that doesn't match any of the native available functions.

- Triggers the **[ValueReceived](#valuereceived)** event if the call was associated with sending value.

- If the data sent is **shorter than 4 bytes**, then the call should pass. For exmaple, sending `0xaabbcc` to the contract will be successfull.

- If the data sent is **preprended with 4 zeros (0)**, then the call should pass.

  For example, when sending value to the contract it can be associated with a message, such as `"Here is 1 Ether"`, after enconding the message as bytes: `0x486572652069732031204574686572`.

  In order to make the call to the contract pass, it should be prepended with `bytes4(0)`. A call to the **LSP0** with `0x00000000486572652069732031204574686572` as data **will pass**.

> - If the data sent is equal or larger to 4 bytes and it's not preprended with `bytes4(0)`, the following steps will be executed:

Reviewer: ```suggestion
- If the data sent is equal or larger to 4 bytes and is not prepended with `bytes4(0)`, the following steps will be executed:
```

>   - Query the storage of the LSP0, and check if there is an address under this following data Key:

Reviewer: Also suggesting this link to ERC725Y subsection on LSP0 page:

```suggestion
  - Query the [storage of the LSP0](../universal-profile/lsp0-erc725account.md#erc725y---generic-key-value-store), and check if there is an address under this following data Key:
```

### transferOwnership

:::info

Check the [**transferOwnership(..)**](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-0-ERC725Account.md#transferownership) function specification in [**LSP0-ERC725Account Standard**](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-0-ERC725Account.md) in the **LIP repository**.

Check the [**transferOwnership(..)**](https://github.com/lukso-network/lsp-smart-contracts/blob/v0.8.0/contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol#L290) function implementation in **LSP0ERC725Account** Contract.

:::

```solidity
function transferOwnership(address newOwner) public
```

Initiate an ownership transfer by setting the `newOwner` as `pendingOwner`.

> Calls the `universalReceiver(..)` function [**on the pending owner**](https://github.com/lukso-network/lsp-smart-contracts/blob/v0.8.0/contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol#L328), if it supports LSP1 InterfaceId with the following typeId:

Reviewer: I suggest that we say `newOwner` for this line, as for new developer it might be easier to understand.

```suggestion
This will also call the `universalReceiver(..)` function [**of the `newOwner`**](https://github.com/lukso-network/lsp-smart-contracts/blob/v0.8.0/contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol#L328), if it supports LSP1 InterfaceId with the following typeId:
```

Author: Actually, i think naming the param `newOwner` is confusing, because it's not the newOwner until it acceptOwnership. So I suggest keeping the pendingOwner name, and renaming the parameter to something like potentielOwner or pendingOwner

### universalReceiver

:::info

Check the [**universalReceiver(..)**](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-0-ERC725Account.md#universalreceiver) function specification in [**LSP0-ERC725Account Standard**](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-0-ERC725Account.md) in the **LIP repository**.

Check the [**universalReceiver(..)**](https://github.com/lukso-network/lsp-smart-contracts/blob/v0.8.0/contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol#L235) function implementation in **LSP0ERC725Account** Contract.

Check the **javascript** guides to know [**How to set a UniversalReceiverDelegate built by LUKSO**](../../guides/universal-receiver-delegate/set-default-implementation.md) or [**How to set your own UniversalReceiverDelegate**](../../guides/universal-receiver-delegate/accept-reject-assets.md).

:::

```solidity
function universalReceiver(
    bytes32 typeId,
    bytes memory data
) public payable returns (bytes memory result)
```

> - Forwards the `msg.data` appended with the `msg.sender` and `msg.value` to the **UniversalReceiverDelegate** contract address stored under the data key attached below, if it supports [LSP1UniversalReceiver InterfaceId](../smart-contracts/interface-ids.md).

Author: I think the best place to have schemas, is in ABI and guides, I am planning to remove the schemas from lsp1 docs page.

```json
{
  "name": "LSP1UniversalReceiverDelegate",
  "key": "0x0cfc51aec37c55a4d0b1a65c6255c4bf2fbdf6277f3cc0730c45b828b6db8b47",
  "keyType": "Singleton",
  "valueType": "address",
  "valueContent": "Address"
}
```

> - Forwards the `msg.data` appended with the `msg.sender` and `msg.value` to the **typeId delegate** contract address stored under the data key attached below, if it supports [LSP1UniversalReceiver InterfaceId](../smart-contracts/interface-ids.md).

Author: https://github.com/lukso-network/docs/pull/461#discussion_r1105509955

#### Return Values:

| Name     | Type    | Description                                                                    |
| :------- | :------ | :----------------------------------------------------------------------------- |
| `result` | `bytes` | The return value of the UniversalReceiverDelegates contracts encoded as bytes. |

### isValidSignature

:::info

Check the [**isValidSignature(..)**](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-0-ERC725Account.md#isvalidsignature) function specification in [**LSP0-ERC725Account Standard**](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-0-ERC725Account.md) in the **LIP repository**.

Check the [**isValidSignature(..)**](https://github.com/lukso-network/lsp-smart-contracts/blob/v0.8.0/contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol#L159) function implementation in **LSP0ERC725Account** Contract.

:::

```solidity
function isValidSignature(
    bytes32 dataHash,
    bytes memory signature
) public view returns (bytes4 magicValue)
```

Checks if a signature was signed by the `owner` of the contract, according to [**EIP-1271**](https://eips.ethereum.org/EIPS/eip-1271).

> If the `owner` is an EOA, the **ECDSA algorithm** will be used to recover the address of the signer from the dataHash and the signature, the function will return the magicValue pointing that the signature is valid, if the address recovered matches the address of the owner. Will return the failure value otherwise.

Reviewer: I suggest we link to EIP1271 here.

I just put the link for `MAGIC VALUE` at the specifications section of EIP1271 as it shows it in the constants.

<img width="628" alt="image" src="https://user-images.githubusercontent.com/31145285/218110982-7b6bd26c-f8ed-464d-8586-3097a8f39e20.png">

The fail value is not explicitly stated in the EIP1271 so my suggestion would be to write `0xffffffff`

```suggestion
If the `owner` is an EOA, the **ECDSA algorithm** will be used to recover the address of the signer from the dataHash and the signature, the function will return the [**`MAGICVALUE`**](https://eips.ethereum.org/EIPS/eip-1271#specification) pointing that the signature is valid, if the address recovered matches the address of the owner. Will return the **`FAIL VALUE`** `0xffffffff` otherwise.
```
